Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/RestShapeSpringsForceField.inl

```diff
@@ -470,18 +472,18 @@ void RestShapeSpringsForceField<DataTypes>::addKToMatrix(const MechanicalParams*
 
         // translation
         const auto vt = -kFact * (index < k.size() ? k[index] : k[0]);
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < space_size; i++)
         {
-            mat->add(offset + N * curIndex + i, offset + N * curIndex + i, vt);
+            mat->add(offset + total_size * curIndex + i, offset + total_size * curIndex + i, vt);
         }
 
         // rotation (if applicable)
-        const auto vr = -kFact * (index < k_a.size() ? k_a[index] : k_a[0]);
         if constexpr (isRigidType<DataTypes>())
         {
-            for (int i = 3; i < 6; i++)
+            const auto vr = -kFact * (index < k_a.size() ? k_a[index] : k_a[0]);
+            for (int i = space_size; i < total_size; i++)
```

Comment:
I think this loop can be automatically optimized away if `space_size==total_size`. Same for `vr`, because it is not used (can the compiler see both of the optimizations?).
Therefore, no need for the `constexpr if`. However, I think it's clearer like that. It separates clearly the rotation from the translation.

Reply:
Agreed